src/main/kotlin/org/jetbrains/research/testspark/helpers/psiHelpers/JavaPsiHelper.kt

```diff
@@ -173,145 +136,313 @@ class JavaPsiHelper : PsiHelperInterface {
+                    " 2) Method ${javaPsiMethodWrapped.methodDescriptor}" +
+                    " 3) Line $line",
+            )
+        }
+
+        return result.toArray()
+    }
+
+    override fun getLineDisplayName(line: Int): String {
+        return "<html><b><font color='orange'>line</font> $line</b></html>"
+    }
+
+    override fun getClassDisplayName(psiClass: PsiClassWrapper): String {
+        return if ((psiClass as JavaPsiClassWrapper).isInterface) {
             "<html><b><font color='orange'>interface</font> ${psiClass.qualifiedName}</b></html>"
-        } else if (isAbstractClass(psiClass)) {
+        } else if (psiClass.isAbstractClass) {
             "<html><b><font color='orange'>abstract class</font> ${psiClass.qualifiedName}</b></html>"
         } else {
             "<html><b><font color='orange'>class</font> ${psiClass.qualifiedName}</b></html>"
         }
     }
 
-    /**
-     * Gets the display name of a method, depending on if it is a (default) constructor or a normal method.
-     * This is used when displaying the name of a method in GenerateTestsActionMethod menu entry.
-     *
-     * @param psiMethod the PSI method of interest
-     * @return the display name of the PSI method
-     */
-    private fun getMethodDisplayName(psiMethod: PsiMethod): String {
-        return if (isDefaultConstructor(psiMethod)) {
+    override fun getMethodDisplayName(psiMethod: PsiMethodWrapper): String {
+        return if ((psiMethod as JavaPsiMethodWrapper).isDefaultConstructor) {
             "<html><b><font color='orange'>default constructor</font></b></html>"
         } else if (psiMethod.isConstructor) {
             "<html><b><font color='orange'>constructor</font></b></html>"
-        } else if (isMethodDefault(psiMethod)) {
+        } else if (psiMethod.isMethodDefault) {
             "<html><b><font color='orange'>default method</font> ${psiMethod.name}</b></html>"
         } else {
             "<html><b><font color='orange'>method</font> ${psiMethod.name}</b></html>"
         }
     }
+
+    private fun validateLine(selectedLine: Int, psiMethod: JavaPsiMethodWrapper, psiFile: PsiFile): Boolean {
+        val doc = PsiDocumentManager.getInstance(psiFile.project).getDocument(psiFile) ?: return false
+        val psiMethodBody: PsiCodeBlock = psiMethod.body ?: return false
+        if (psiMethodBody.statements.isEmpty()) return false
+
+        val firstStatement = psiMethodBody.statements.first()
+        val lastStatement = psiMethodBody.statements.last()
+
+        val firstStatementLine = doc.getLineNumber(firstStatement.startOffset)
+        val lastStatementLine = doc.getLineNumber(lastStatement.endOffset)
+
+        return selectedLine in firstStatementLine..lastStatementLine
+    }
+
+    private fun withinElement(psiElement: PsiElement, caretOffset: Int): Boolean {
+        return (psiElement.textRange.startOffset <= caretOffset) && (psiElement.textRange.endOffset >= caretOffset)
```

Comment:
**[suggestion]** May be implemented as a extension method of `PsiElement` and renamed to `containsOffset`:

```kotlin
fun PsiElement.containsOffset(offset: Int): Bool { .. }
```